blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/jira/JiraSCMListener.java

```diff
@@ -0,0 +1,74 @@
+package io.jenkins.blueocean.service.embedded.jira;
+
+import com.atlassian.jira.rest.client.api.domain.Issue;
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import hudson.Extension;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model.listeners.SCMListener;
+import hudson.plugins.jira.JiraBuildAction;
+import hudson.plugins.jira.JiraSession;
+import hudson.plugins.jira.JiraSite;
+import hudson.plugins.jira.model.JiraIssue;
+import hudson.scm.ChangeLogSet;
+import hudson.scm.SCM;
+
+import java.util.List;
+import java.util.Set;
+
+@Extension
+public class JiraSCMListener extends SCMListener {
+    @Override
+    public void onChangeLogParsed(Run<?, ?> run, SCM scm, TaskListener listener, ChangeLogSet<?> changelog) throws Exception {
+        JiraSite jiraSite = JiraSite.get(run.getParent());
+        if (jiraSite == null) {
+            return;
+        }
+        List<String> issueKeys = Lists.newArrayList();
+        for (ChangeLogSet.Entry entry : changelog) {
+            issueKeys.addAll(BlueJiraIssue.findIssueKeys(entry.getMsg(), jiraSite.getIssuePattern()));
```

Comment:
Switched to using a LinkedHashSet. BlueJiraIssue has toEquals and hashCode implemented